include/miopen/config.h.in

```diff
@@ -78,12 +78,55 @@
 
 // clang-format off
 #cmakedefine MIOPEN_DEFAULT_FIND_MODE @MIOPEN_DEFAULT_FIND_MODE@
+
 // clang-format on
+#ifndef MIOPEN_CMP_FLAGS_NEW
 #cmakedefine MIOPEN_AMDGCN_ASSEMBLER "@MIOPEN_AMDGCN_ASSEMBLER@"
 #cmakedefine HIP_OC_COMPILER "@HIP_OC_COMPILER@"
 #cmakedefine MIOPEN_HIP_COMPILER "@MIOPEN_HIP_COMPILER@"
 #cmakedefine MIOPEN_OFFLOADBUNDLER_BIN "@MIOPEN_OFFLOADBUNDLER_BIN@"
 #cmakedefine MIOPEN_CACHE_DIR "@MIOPEN_CACHE_DIR@"
+#else /*MIOPEN_CMP_FLAGS_NEW if defined*/
+/* Include rocm-core header for get ROCm Install Path Method */
+#include <rocm-core/rocm_getpath.h>
+
+/* Flags to help identify/differentiate Definitions for each MIOPEN Compiler Flags */
+typedef enum {
+        MIOPEN_AMDGCN_Flag=0,
+        MIOPEN_OC_COMPILER_Flag=1,
+        MIOPEN_HIP_COMPILER_Flag=2,
+        MIOPEN_OFFLOAD_BUNDLER_Flag=3,
+        MIOPEN_ROCM_INSTALL_PATH_Flag=4,
+        MIOPEN_CMP_MAX_Flag=5 /*Should be last Flag definition*/
+} MIOPEN_CMP_Flags_t;
+
+/* List to hold Relative Directory Path for each Compiler Flags from ROCM Install Path */
+extern const char* MIOPEN_CMP_FLAG_REL_PATHS[MIOPEN_CMP_MAX_Flag];
+
+/* List to hold the Compiler Flag Paths*/
+extern char MIOPEN_CMP_FLAGS[MIOPEN_CMP_MAX_Flag][PATH_MAX];
+
+/* Macro to initialize the MIOPEN Compiler Flags */
+#define MIOPEN_CMP_FLAGS_INIT() do{char *flagPath=NULL; int len = 0;  MIOPEN_CMP_Flags_t i;\
+        if(0==MIOPEN_CMP_FLAGS[MIOPEN_AMDGCN_Flag][0]){\
+          PathErrors_t ret = getROCmInstallPath(&flagPath, &len); \
+          if( PathSuccess == ret ){\
+            for(i=MIOPEN_AMDGCN_Flag;i<MIOPEN_CMP_MAX_Flag;i++){\
+                  sprintf((char *)*(MIOPEN_CMP_FLAGS+i),"%s%s", flagPath, MIOPEN_CMP_FLAG_REL_PATHS[i] );\
```

Comment:
No limit on the length? There are a couple of of approaches, just misbehave of the strings are too long or else limit the lengths, probably using snprintf. The problem is what to do if the results are truncated?

I think
     sprintf(&(MIOPEN_CMP_FLAGS[i]),"%s/%s", flagPath,MIOPEN_CMP_FLAG_REL_PATHS[i])
   is a bit more readable.

I think this code could go in a function rather than a macro.

Reply:
>>No limit on the length? There are a couple of of approaches, just misbehave of the strings are too long or else limit the lengths, probably using snprintf. The problem is what to do if the results are truncated? 
There is a length check added in later versions, Kindly take a look from latest commit. 

>>I think
sprintf(&(MIOPEN_CMP_FLAGS[i]),"%s/%s", flagPath,MIOPEN_CMP_FLAG_REL_PATHS[i])
is a bit more readable. 
Thanks for the comments, will work on this comments

Reply:
Updated in latest version https://github.com/ROCm/MIOpen/pull/2694/commits/f503776de6a0a07aa4408b1c48c56d524bc0638a